File: fix_icons.js
const fs = require('fs');
const glob = require('glob');

const files = glob.sync('src/components/**/*.tsx');
files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let original = content;
  
  // Clean up fill-white where it breaks icons in dark mode
  content = content.replace(/className=\"[^\"]*fill-white text-foreground size-5[^\"]*\"/g, 'className=\"text-white size-5\"');
  content = content.replace(/<Eye className=\"fill-white size-7 \" \/>/g, '<Eye className=\"text-white size-5\" />');
  content = content.replace(/<PencilLine className=\"fill-white text-white\" \/>/g, '<PencilLine className=\"text-white\" />');
  content = content.replace(/<FaRegTrashAlt className=\"text-white fill-white size-4 \" \/>/g, '<FaRegTrashAlt className=\"text-white size-4\" />');
  content = content.replace(/<IoEye className=\"size-6 fill-white\" \/>/g, '<IoEye className=\"size-5 text-white\" />');
  content = content.replace(/<Square className=\"size-4 fill-white\" \/>/g, '<Square className=\"size-4 text-white\" />');
  content = content.replace(/<Play className=\"size-3 fill-white\" \/>/g, '<Play className=\"size-3 text-white\" />');
  content = content.replace(/<Eye className=\"fill-white text-foreground size-5\" \/>/g, '<Eye className=\"text-white size-5\" />');
  
  if (content !== original) {
    fs.writeFileSync(file, content);
    console.log('Fixed', file);
  }
});

File: fix_icons.py
import os
import glob
import re

files = glob.glob('src/components/**/*.tsx', recursive=True)
count = 0
for filepath in files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    original = content
    
    # 1. CirclePlus
    content = re.sub(r'className="fill-white text-foreground size-5"', 'className="text-white size-5"', content)
    
    # 2. Eye
    content = re.sub(r'className="fill-white size-7 "', 'className="text-white size-5"', content)
    
    # 3. FaRegTrashAlt
    content = re.sub(r'className="text-white fill-white size-4 "', 'className="text-white size-4"', content)
    
    # 4. PencilLine
    content = re.sub(r'className="fill-white text-white"', 'className="text-white size-5"', content)
    
    # 5. IoEye
    content = re.sub(r'className="size-6 fill-white"', 'className="size-5 text-white"', content)
    
    # 6. MdDataSaverOn
    content = re.sub(r'className="size-5 fill-\[#1797B9\]/90 hover:fill-white transition-all duration-300"', 'className="size-5 text-[#1797B9]/90 hover:text-white transition-all duration-300"', content)

    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Fixed {filepath}")
        count += 1

print(f"Total files fixed: {count}")
